Source/WTF/wtf/text/ASCIILiteral.h

```diff
@@ -143,19 +140,18 @@ constexpr ASCIILiteral operator"" _s(const char* characters, size_t)
     return result;
 }
 
-constexpr std::span<const LChar> operator"" _span(const char* characters, size_t n)
+constexpr std::span<const LChar> operator"" _span(const char* characters, size_t)
 {
+    auto span = unsafeForgeSpan(bitwise_cast<const LChar*>(characters), std::char_traits<char>::length(characters));
```

Comment:
I think this inserts a new call to strlen() when the length was otherwise known at compile time, right? I'd prefer to use 'n' here instead, for performance, and also it's kinda safer? Because we know it's the 'n' we got from the compiler's constant string.

Reply:
(I've suggested, separately to clang developers, that the warning should improve not even to warn about 'n' in this case, for the same reason.)

Reply:
Oof.  If `std::char_traits<char>::length()` calls `strlen()` at runtime, we're paying that cost for all of our constant strings using the `_s` operator!

I can fix that, though.

Reply:
Just to be sure of my reading of the docs, I double-checked:

```
~/scratch> cat ./scratch.cc                                                               
#include <memory>
#include <array>
#include <span>
#include <string>

using namespace std;

size_t f(const char* characters)
{
	return std::char_traits<char>::length(characters);
}

int main()
{
}
~/scratch> clang++ -std=c++23 -O3 -o scratch scratch.cc  && otool -tvV ./scratch | c++filt
./scratch:
(__TEXT,__text) section
f(char const*):
0000000100003f90	b	0x100003f9c ; symbol stub for: _strlen
_main:
0000000100003f94	mov	w0, #0x0
0000000100003f98	ret

```

Reply:
In the constant case, if you get sufficient inlining, the call to strlen disappears. So not a disaster. But not optimal.

Reply:
Latest PR fixed both `_span` as well as `_s` operators.

Reply:
Of course, that fix triggered buggy behavior in gcc that can't decide between `String::String(std::span<const char>)` constructor and `String::String(ASCIILiteral)` with `ASCIILiteral::ASCIILiteral(std::span<const char>)` constructor, even though the `ASCIILiteral` constructor is `private` and not even available to use.